bless/aws_lambda/bless_lambda.py

```diff
@@ -57,13 +59,86 @@ def lambda_handler(event, context=None, ca_private_key_password=None,
     config = BlessConfig(region,
                          config_file=config_file)
 
-    logging_level = config.get(BLESS_OPTIONS_SECTION, LOGGING_LEVEL_OPTION)
-    numeric_level = getattr(logging, logging_level.upper(), None)
-    if not isinstance(numeric_level, int):
-        raise ValueError('Invalid log level: {}'.format(logging_level))
+    logger = set_logger(config)
 
-    logger = logging.getLogger()
-    logger.setLevel(numeric_level)
+    ca_private_key = config.getprivatekey()
+    password_ciphertext_b64 = config.getpassword()
+
+    # Process cert request
+    schema = BlessHostSchema(strict=True)
+
+    try:
+        request = schema.load(event).data
+    except ValidationError as e:
+        return error_response('InputValidationError', str(e))
+
+    logger.info('Bless lambda invoked by [public_key: {}]'.format(request.public_key_to_sign))
+
+    # Decrypt ca private key password
+    if ca_private_key_password is None:
+        try:
+            ca_private_key_password = decrypt_ca_password(password_ciphertext_b64)
+        except ClientError as e:
+            return error_response('ClientError', str(e))
+
+    # if running as a Lambda, we can check the entropy pool and seed it with KMS if desired
+    if entropy_check:
+        check_entropy(config, logger)
+
+        # cert values determined only by lambda and its configs
+    current_time = int(time.time())
+    valid_before = current_time + int(timedelta(days=365).total_seconds())  # Host certificate is valid for at least 1 year
+    valid_after = current_time
+
+    # Build the cert
+    ca = get_ssh_certificate_authority(ca_private_key, ca_private_key_password)
+    cert_builder = get_ssh_certificate_builder(ca, SSHCertificateType.HOST,
+                                               request.public_key_to_sign)
+
+    cert_builder.set_valid_before(valid_before)
+    cert_builder.set_valid_after(valid_after)
+
+    # cert_builder is needed to obtain the SSH public key's fingerprint
+    key_id = 'request[{}] ssh_key[{}]  ca[{}] valid_to[{}]'.format(
+        context.aws_request_id, cert_builder.ssh_public_key.fingerprint, context.invoked_function_arn,
+        time.strftime("%Y/%m/%d %H:%M:%S", time.gmtime(valid_before))
+    )
+    cert_builder.add_valid_principal(request.hostnames)
+    cert_builder.set_key_id(key_id)
+    cert = cert_builder.get_cert_file()
+
+    logger.info(
+        'Issued a cert to hostnames[{}] with key_id[{}] and '
+        'valid_from[{}])'.format(
+            request.hostnames, key_id,
+            time.strftime("%Y/%m/%d %H:%M:%S", time.gmtime(valid_after))))
+    return success_response(cert)
+
+
+def lambda_handler_user(
```

Comment:
Renaming the lambda_handler_user is fine, but we'll want to keep lambda_handler as a wrapper to lambda_handler_user for seamless upgrades.